### ISSUE TITLE
missing 'template' in Autocharge-optimal definition

### ISSUE THREAD

**Issue by lepricon** [top-gun/Tesla-PV-charging#3]
The definition on autocharge-optimal is not proper yaml, since is starting from the - sign. Probably 'template' keyword is missing. Which "include" statement you have above the pasted content?

**Comment by lepricon** [comment]
Which "include" statement you have above the pasted content?

**Comment by silencer2008** [comment]
@lepricon did you manage to solve this ? having the same issue of entity not defined.

**Comment by lepricon** [comment]
Yep, divided it into two files: template.yaml and sensor.yaml and thus it concatenates correctly

**Comment by silencer2008** [comment]
would you be able to share it please ? still having issues

**Comment by top-gun** [comment]
> The definition on autocharge-optimal is not proper yaml, since is starting from the - sign. Probably 'template' keyword is missing. Which "include" statement you have above the pasted content?

There is no need to seperate this into several files.

Sensors are generally a long list. If this is your only one, it needs to be preceded by "template:" like this:

**template:**
  - sensor:
    - name: 'Autocharge-optimal'
      unit_of_measurement: "A"
      state: > 
        {# calculate the optimal charge current based on several parameters: #}
        {# PV yield, house battery SOC, vehicle battery SOC, Grid consumption #}
...
